driver/reduce_driver.hpp

```diff
@@ -493,20 +493,20 @@ int ReduceDriver<Tgpu, Tref>::VerifyForward()
     if(std::is_same<Tgpu, float>::value && reduceOp == MIOPEN_REDUCE_TENSOR_NORM2)
         tolerance *= 12.0;
 
-    if(error > tolerance)
+    if(!std::isfinite(error) || error > tolerance)
     {
-        std::cout << "ReduceTensor() Failed with error = " << error
-                  << " , tolerance = " << tolerance << "\n";
+        std::cout << "ReduceTensor() FAILED with error = " << error
+                  << " , tolerance = " << tolerance << std::endl;
     }
     else
     {
         if(out_indices.size() > 0)
         {
             auto error2 = miopen::rms_range(outhost_indices, out_indices);
 
-            if(static_cast<float>(error2) != 0.0f)
```

Comment:
The tolerance checking could not be used with error2, since error2 is the rms_range of two vector<int> representing the outputs of indices by Host and GPU reduction.  For outputs of indices, we want the GPU result be completely the same as the Host result.  
We can use **if(!std::isfinite(error2) && static_cast<float>(error2) != 0.0f )**